CDS_retriever.py

```diff
@@ -132,31 +134,121 @@ def year_retrieve(dataset, var, freq, year, grid, levelout, area, outdir, reques
                 os.remove(f)
 
 
-# define propertes for vertical levels
+
+plevs = {'ERA5' : ['1000', '975', '950', '925',
```

Comment:
did not try this, but this is dictionary based on the dataset. Did you modified the way this is read? At L246 it does not seem to me you are accessing `levels['ERA5']`... of course I might be wrong!

Reply:
I slightly modified the logic of the function. Your 'define_level' assigns 'level' and 'level kind' based on a series of switches, while mine does the same but also includes a check for custom pressure levels. It verifies whether the request is for predefined levels (stored in the 'plevs' dictionary) or for (one or multiple) custom pressure levels, checking their validity against the 'plevs['ERA5']' list.

The main difference is the validation of arbitrary pressure levels, which I handle at line 199. Additionally, users can input arbitrary levels with or without the unit of measure (hPa only, for now, as per the regex at line 190).

I also generally prefer dictionary searches over using if blocks.

Reply:
Now I undersand `plevs['ERA5'] is the FULL set of levels for ERA5